h2o-algos/src/test/java/hex/ensemble/StackedEnsembleTest.java

```diff
@@ -1098,5 +1099,66 @@ public void testInvalidFoldColumn_trainingFrame() {
     }
   }
 
+  @Test
+  public void testBaseModelsWorkWithGrid() {
+    GBMModel gbmModel = null;
+    try {
+      Scope.enter();
+
+      final Frame trainingFrame = TestUtil.parse_test_file("./smalldata/junit/weather.csv");
+      Scope.track(trainingFrame);
+      trainingFrame.toCategoricalCol("RainTomorrow");
+
+      GBMModel.GBMParameters parameters = new GBMModel.GBMParameters();
+      parameters._train = trainingFrame._key;
+      parameters._seed = 0xFEED;
+      parameters._response_column = "RainTomorrow";
+      parameters._ntrees = 1;
+      parameters._keep_cross_validation_predictions = true;
+
+      GBM gbm = new GBM(parameters);
+      gbmModel = gbm.trainModel().get();
+      assertNotNull(gbmModel);
+
+      final Integer[] ntreesArr = new Integer[]{5, 50, 7, 8, 9, 10};
+      final Integer[] maxDepthArr = new Integer[]{2, 3, 4};
+      HashMap<String, Object[]> hyperParms = new HashMap<String, Object[]>() {{
+        put("_distribution", new DistributionFamily[]{DistributionFamily.bernoulli});
+        put("_ntrees", ntreesArr);
+        put("_max_depth", maxDepthArr);
+      }};
+
+      GBMModel.GBMParameters params = new GBMModel.GBMParameters();
+      params._train = trainingFrame._key;
+      params._response_column = "RainTomorrow";
+      params._seed = 42;
 
+      Job<Grid> gs = GridSearch.startGridSearch(null, params, hyperParms);
+      Scope.track_generic(gs);
+      final Grid grid = gs.get();
+      Scope.track_generic(grid);
+
+      final StackedEnsembleParameters seParams = new StackedEnsembleParameters();
+      seParams._train = trainingFrame._key;
+      seParams._response_column = "RainTomorrow";
+      seParams._metalearner_algorithm = Algorithm.AUTO;
+      seParams._base_models = new Key[]{gbmModel._key, grid._key};
+      seParams._seed = 0xFEED;
+
+      final StackedEnsemble stackedEnsemble = new StackedEnsemble(seParams);
+      List<Key> expectedBaseModels = new ArrayList<Key>();
+      expectedBaseModels.add(gbmModel._key);
+      Collections.addAll(expectedBaseModels, grid.getModelKeys());
+
+      assertArrayEquals(expectedBaseModels.toArray(new Key[0]), stackedEnsemble._parms._base_models);
```

Comment:
am I missing sth here? 
this SE hasn't been trained, so `stackedEnsemble._parms._base_models` should be just equal to `seParams._base_models = new Key[]{gbmModel._key, grid._key}`.
then you can train the SE model, and check the `base_models` attribute on the SE, not on the `params`:
```
assertArrayEquals(expectedBaseModels.toArray(new Key[0]), stackedEnsemble.base_models);
```

Reply:
I am probably missing here something. The test I wrote works because it expands the `_parms._base_models` during `StackedEnsemble.init`. I didn't find any attribute `StackedEnsemble.base_models` or `StackedEnsembleModel.base_models` etc. and wherever I looked the `_parms._base_models` was used so I used it as well.

The only `base_model` attribute I could find is in schemas and I don't know yet how to get to it after the model was trained. Any hints?

Reply:
forgot about it, must be tired... for a second I thought this was a client test...